src/pages/AdminPages/Transactions/PendingTransactions.jsx

```diff
@@ -0,0 +1,129 @@
+
+        const transactionResponse = {
+            processedTransactions,
+            pendingTransactions,
+            data : {
+                ...selectedAlterContent,
+                reason,
+                status
+            }
+        }
+
+        dispatch(sendTransactionResponse(transactionResponse))
+        dispatch(disableButton())
+
+        closeShowResponse()
+
+       
+    }
+
+
+    return(
+        <>
+
+
+                        <div>Pending Transactions</div>
+                        {
+                            pendingTransactions && (
+                                pendingTransactions.length > 0 ?
+
+                                <>
+
+                                    <BaseModal
+                                        title="Transaction Response"
+                                        show={showResponse}
+                                        close={() => closeShowResponse()}
+                                        closeText="Close"
+                                        size="sm"
+                                    >
+                                        
+                                        <Form
+                                            initialValues={{ reason : "", status: "declined" }}
+                                            onSubmit={processTransaction}
+                                            validationSchema={validationSchema}
+                                        >
+
+
+                                            <TextareaField 
+                                                name="reason"
+                                            />
+
+                                            <SubmitButton title="Decline" disable={buttonDisable} />
+
+                                            <button onClick={() => processTransaction({ reason : "", status : "approved"})}>Approve</button>
+
+                                        </Form>
+
+
+                                    </BaseModal>
+
+
+                                    <Table 
+                                        headers={[ ...header, {dataKey:"action", headerText:"Action"} ]}
+                                        data={pendingTransactions.map(trans => ({
+                                                ...trans,
+                                                'time' : timeAgo(trans.time - TIME_ZONE),
+                                                'amount' :  `N${thousands_separators(trans.amount.toFixed(2))}`,
+                                                'status' : "Pending",
+                                                'name' : `${trans.firstName} ${trans.lastName}`,
+                                                'receipt' : trans.txType === "credit" ? 
+                                                    <>
+                                                        <span onClick={() => setContent("Uploaded Receipt", "md", <img alt="reciept" src={trans.receipt} />) }>REciept</span>
```

Comment:
Why not make it go to another page or it should be automatically downloaded?

Reply:
Not sure why you'd want that... This creates a way better User Experience than making it download or load a different page entirely.  A popup is pretty better in this case, it won't disrupt the flow of activity.